Data/models/Bookmark.swift

```diff
@@ -196,17 +196,21 @@ public final class Bookmark: NSManagedObject, WebsitePresentable, Syncable, CRUD
             bk.domain = Domain.getOrCreateForUrl(url, context: context, save: false)
         }
         
-        // This also sets up a parent folder
-        bk.syncParentUUID = bookmark?.parentFolderObjectId ?? bk.syncParentUUID
+        // Update parent folder if one exists
+        if let newParent = bookmark?.parentFolderObjectId {
+            bk.syncParentUUID = newParent
+        }
         
         // For folders that are saved _with_ a syncUUID, there may be child bookmarks
         //  (e.g. sync sent down bookmark before parent folder)
         if bk.isFolder {
             // Find all children and attach them
-            if let children = Bookmark.getNonFolderChildren(forFolderUUID: bk.syncUUID) {
-                
-                // TODO: Setup via bk.children property instead
-                children.forEach { $0.syncParentUUID = bk.syncParentUUID }
+            if let children = Bookmark.getChildren(forFolderUUID: bk.syncUUID, context: context) {
+                // Re-link all orphaned children
+                children.forEach {
+                    $0.syncParentUUID = bk.syncUUID
+                    $0.parentFolder = bk
```

Comment:
Definitely should add a comment here:

$0.syncParentUUID = bk.syncUUID
// The `setter` for `syncParentUUID` creates the parent/child relationship in CD, however in this specific instance
//  the objects have not been written to disk, so cannot be fetched on a different context and the relationship
//  will not be properly established. Manual attachment is necessary here during these batch additions.
$0.parentFolder = bk

Reply:
Thanks, any suggestion to add comments are really appreciated by me :)

Reply:
This is resolved btw